tests/integration/src/test/java/org/mule/NonBlockingFunctionalTestCase.java

```diff
@@ -0,0 +1,204 @@
+package org.mule;
+
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleEventContext;
+import org.mule.api.MuleMessage;
+import org.mule.construct.Flow;
+import org.mule.tck.functional.EventCallback;
+import org.mule.tck.functional.FlowAssert;
+import org.mule.tck.functional.FunctionalTestComponent;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.util.concurrent.Latch;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+public class NonBlockingFunctionalTestCase extends FunctionalTestCase
+{
```

Comment:
The problem with this kind of tests is that you're loading the entire config file (with several forms) in each test, but each test only uses just one flow. This will impact in the build time. How about to separate this case into several cases using a smaller config for each one?